File: app/src/main/java/com/sha/formvalidatorsample/ui/MainActivity.java
package com.sha.formvalidatorsample.ui;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;

import com.sha.formvalidatorsample.R;

public class MainActivity extends Activity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        findViewById(R.id.btnFields).setOnClickListener(__ -> show(FieldsActivity.class));
        findViewById(R.id.btnForm).setOnClickListener(__ -> show(FormActivity.class));
        findViewById(R.id.btnRxForm).setOnClickListener(__ -> show(RxFormActivity.class));
    }

    private void show(Class clazz) {
        startActivity(new Intent(this, clazz));
    }

}

File: sample/src/main/java/com/sha/formvalidatorsample/classic/ui/MainActivity.kt
package com.sha.formvalidatorsample.classic.ui

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.view.View

import com.sha.formvalidatorsample.R
import com.sha.formvalidatorsample.compose.ComposeActivity

class MainActivity : Activity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        findViewById<View>(R.id.btnFields).setOnClickListener { show(FieldsActivity::class.java) }
        findViewById<View>(R.id.btnForm).setOnClickListener { show(FormActivity::class.java) }
        findViewById<View>(R.id.btnCompose).setOnClickListener { show(ComposeActivity::class.java) }
    }

    private fun show(clazz: Class<*>) = startActivity(Intent(this, clazz))

}
